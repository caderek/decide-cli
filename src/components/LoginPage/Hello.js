import React from 'react'
import $ from './Hello.scss'

const Hello = () => (
  <div className={$.hello}>
    <img src='logo.png' alt='Logo' />
    <h2>Prioritize your{'\u00A0'}ideas with your{'\u00A0'}team.</h2>
  </div>
)

export default Hello
