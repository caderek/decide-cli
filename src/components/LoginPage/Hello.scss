.hello {
  text-align: center;

  h2 {
    color: #70889C;
    margin-top: 15px;
    margin-bottom: 80px;
  }
}
