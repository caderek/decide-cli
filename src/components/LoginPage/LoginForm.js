import React, { Component } from 'react'
import socket from '../../socket'
import $ from './LoginForm.scss'

class LoginForm extends Component {
  constructor (props) {
    super(props)
    this.state = {
      user: '',
      password: ''
    }

    this.handleChange = this.handleChange.bind(this)
    this.authenticate = this.authenticate.bind(this)
  }

  authenticate (event) {
    event.preventDefault()

    socket.emit('authenticate', {
      user: this.state.user,
      password: this.state.password
    })
  }

  handleChange ({ target }) {
    const field = target.type === 'text' ? 'user' : 'password'
    this.setState({ [field]: target.value })
  }

  render () {
    return (
      <form className={$.loginForm} onSubmit={this.authenticate} >
        <input
          type='text'
          placeholder='user'
          value={this.state.user}
          onChange={this.handleChange}
          autoFocus
        />
        <input
          type='password'
          placeholder='password'
          value={this.state.password}
          onChange={this.handleChange}
        />
        <input type='submit' value='log in' />
      </form>
    )
  }
}

export default LoginForm
