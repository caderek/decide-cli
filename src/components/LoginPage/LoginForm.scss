.loginForm {
  max-width: 330px;
  margin: auto;

  input[type=text], input[type=password] {
    padding: 10px;
    font-size: 24px;
    font-family: "Fira Code";
    margin-bottom: 10px;
    width: 100%;
    border: solid 1px #C0C1C0;
    outline: none;

    &:focus {
      border-color: #F6A623;
    }
  }

  input[type=submit] {
    font-size: 24px;
    font-family: "Fira Code";
    background-color: #72889A;
    color: #F8F8F8;
    padding: 10px;
    width: 100%;
    outline: none;
    cursor: pointer;

    &:hover {
      background-color: #5B6D7D;
    }
  }
}

