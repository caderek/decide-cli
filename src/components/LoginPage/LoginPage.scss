.loginPage {
  height: 100vh;
  padding: 10px;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}
