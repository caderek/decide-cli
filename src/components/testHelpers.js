import renderer from 'react-test-renderer'

export const snap = component => {
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
}
