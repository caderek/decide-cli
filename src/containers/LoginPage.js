import { connect } from 'react-redux'
import LoginPage from '../components/LoginPage/LoginPage'

const mapStateToProps = (state) => ({
  open: state.authentication.panel
})

export default connect(mapStateToProps)(LoginPage)
