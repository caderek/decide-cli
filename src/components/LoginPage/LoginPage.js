import React from 'react'
import LoginForm from './LoginForm'
import Hello from './Hello'
import $ from './LoginPage.scss'

const LoginPage = ({ authenticated }) => (
  <div className={$.loginPage} style={{
    display: authenticated ? 'none' : 'flex'
  }}>
    <div>
      <Hello />
      <LoginForm />
    </div>
  </div>
)

export default LoginPage
