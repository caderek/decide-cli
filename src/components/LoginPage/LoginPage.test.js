import React from 'react'
import { snap } from '../testHelpers'
import LoginPage from './LoginPage'

test('renders correctly', () => snap(
  <LoginPage />
))
