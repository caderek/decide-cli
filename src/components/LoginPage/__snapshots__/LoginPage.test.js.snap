exports[`test renders correctly 1`] = `
<div
  className="loginPage">
  <div>
    <div
      className="hello">
      <img
        alt="Logo"
        src="logo.png" />
      <h2>
        Prioritize your
         
        ideas with your
         
        team.
      </h2>
    </div>
    <div
      className="loginForm">
      <input
        placeholder="user"
        type="text" />
      <input
        placeholder="password"
        type="password" />
      <button>
        log in
      </button>
    </div>
  </div>
</div>
`;
